### ISSUE TITLE
CloudFormation template has wrong value for "Effect" for AuthRole and UnauthRole

### ISSUE THREAD

**Issue by yaquawa** [aws-amplify/amplify-cli#11839]
### Amplify CLI Version

10.6.1

### What operating system are you using?

MacOS

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

x

### Describe the bug

I'm facing the same problem with #6022
I don't know why it was closed.

### Expected behavior

x

### Reproduction steps

x

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by josefaidt** [comment]
Hey @yaquawa :wave: thanks for raising this! From the [previous issue](https://github.com/aws-amplify/amplify-cli/issues/6022#issuecomment-744639373):

> You're seeing this since we use a custom lambda resource to update the auth and unuath role and change the effect to "Allow" when you have auth enabled in your Amplify project

Are you seeing any impact to your project with this drift present? Regardless, I will mark this as a bug for it to be addressed

**Comment by yaquawa** [comment]
@josefaidt 
> Are you seeing any impact to your project with this drift present?

No, but I just found this issue when I performed a drift detection.
Thanks for marking this as a bug.
